modules/300-prometheus/images/alerts-receiver/src/memstore.go

```diff
@@ -64,41 +72,85 @@ func (a *memStore) insertAlert(alert *model.Alert) {
 		ta.Timeout = true
 		ta.EndsAt = now.Add(resolveTimeout)
 	}
-	fingerprint := ta.Fingerprint()
 
-	if _, ok := a.alerts[fingerprint]; ok {
-		log.Infof("alert with fingerprint %s updated in queue", fingerprint)
-		a.alerts[fingerprint] = ta.Merge(a.alerts[fingerprint])
-	} else {
+	fingerprint := fingerprintWithoutSeverity(alert)
+
+	al, ok := a.alerts[fingerprint]
+	if !ok {
+		if len(a.alerts) == a.capacity {
+			return fmt.Errorf("cannot add alert to queue (capacity = %d), queue is full", a.capacity)
+		}
 		log.Infof("alert with fingerprint %s added to queue", fingerprint)
+		a.alerts[fingerprint] = ta
+		return nil
 	}
-	a.alerts[fingerprint] = ta
 
-	return
-}
+	if al.Labels[severityLabel] > ta.Labels[severityLabel] {
```

Comment:
shouldn't it be vice versa as, according to our severity levels gradation, higher numbers imply lower priorities?

Reply:
yep, thanks !